components/automate-ui/src/app/components/resource-dropdown/resource-dropdown.component.ts

```diff
@@ -0,0 +1,132 @@
+import { Component, EventEmitter, Input, Output, OnChanges, OnInit, SimpleChanges } from '@angular/core';
+
+import { ChefSorters } from 'app/helpers/auth/sorter';
+
+export interface ResourceChecked {
+  id: string;
+  name: string;
+  checked: boolean;
+}
+
+export interface ResourceCheckedMap {
+  [id: string]: ResourceChecked;
+}
+
+@Component({
+  selector: 'app-resource-dropdown',
+  templateUrl: './resource-dropdown.component.html',
+  styleUrls: ['./resource-dropdown.component.scss']
+})
+
+export class ResourceDropdownComponent implements OnInit, OnChanges {
+
+  // The map of resources to operate on.
+  @Input() resources: ResourceChecked[] = [];
+
+  // (Optional) Used to re-synchronize summary label if the set of checked items has changed.
+  // This input is needed only when re-displaying the resource dropdown
+  // for *additional* resources, as with the create-object-modal-component.
+  // Consumers that use it only for a single resource, e.g., team-details.component, should omit.
+  @Input() resourcesUpdated: EventEmitter<boolean>;
+
+  // (Optional) Label to use with no resources selected; defaults to given value if not provided.
+  @Input() noneSelectedLabel = 'None';
+
+  // Plural display name of resource (e.g. "policies", "projects").
+  @Input() objectNounPlural = 'MISSING REQUIRED PARAMETER';
```

Comment:
To emphasize this is a required parameter, give it a very visibly wrong value indicative of what is needed.